### ISSUE TITLE
support SPMD operations like psum, axis_index in jax.vmap (feature request)

### ISSUE THREAD

**Issue by dmjlm** [google/jax#2334]
Feature request for jax.vmap to support SPMD operations like those supported by jax.pmap, namely jax.lax.psum, jax.lax.axis_index, etc.

A few use-cases:
1. Use the same API as pmap without jit.
2. Implementing batch-norm within an un-batched network layer.

Potentially others!

**Comment by mattjj** [comment]
@apaszke squashed this in #4005 and follow-ups! Until #4038 lands you've got to enable omnistaging (#3370) to use axis names with vmap:

```python
from jax.config import config
config.enable_omnistaging()
```
